fanficfare/adapters/adapter_readonlymindcom.py

```diff
@@ -0,0 +1,193 @@
+        of chapters.
+        """
+        if not (self.is_adult or self.getConfig("is_adult")):
+            raise exceptions.AdultCheckRequired(self.url)
+
+        data1 = self.get_request(self.url)
+        logger.debug(self.url)
+
+        soup1 = self.make_soup(data1)
+        #strip comments from soup
+        baseUrl = "https://" + self.getSiteDomain()
+        if 'Page Not Found.' in data1:
+            raise exceptions.StoryDoesNotExist(self.url)
+
+        # Extract metadata
+        header = soup1.find('header')
+        title = header.find('h1')
+        self.story.setMetadata('title', title.text)
+
+        # Author
+        author = soup1.find('meta', attrs={"name":"author"})
+        authorurl = soup1.find('link', rel="author")
+        self.story.setMetadata('author', author.attrs["content"])
+        self.story.setMetadata('authorUrl', baseUrl + authorurl["href"])
+        self.story.setMetadata('authorId', author.attrs["content"])
+
+        # Description
+        synopsis = soup1.find('meta', attrs={"name":"description"})
+        self.story.setMetadata('description', synopsis.attrs["content"])
+
+        # Tags
+        # As these are the only tags should they go in categories?
+        for a in soup1.find_all('a', class_="tag-link"):
+            self.story.addToList('eroticatags', a.text.strip('#').replace('_', ' '))
+
+        # Publish and update dates
+        publishdate = soup1.find('meta', attrs={"name":"created"})
+        pDate = makeDate(publishdate.attrs['content'], self.dateformat)
+        if publishdate is not None: self.story.setMetadata('datePublished', pDate)
+        # if updatedate is not None: self.story.setMetadata('dateUpdated', updatedate)
+
+        # Get chapter URLs
+        chapterTable = soup1.find('section', id='chapter-list')
+        #
+        if chapterTable is not None:
+            # Multi-chapter story
+            chapterRows = chapterTable.find_all('section', class_='story-card-large')
+            for row in chapterRows:
+                titleDiv = row.find('div', class_='story-card-title')
+                chapterCell = titleDiv.a
+                if chapterCell is not None:
+                    chapterTitle = chapterCell.text
+                    chapterUrl = baseUrl + chapterCell['href']
+                    self.add_chapter(chapterTitle, chapterUrl)
+                dateUpdated = row.find('div', class_='story-card-publication-date')
+                if dateUpdated is not None:
+                    self.story.setMetadata('dateUpdated', makeDate(dateUpdated.text, self.dateformat))
+
+        else:
+            # Single chapter
+            chapterTitle = self.story.getMetadata('title')
+            chapterUrl = self.url
+            self.add_chapter(chapterTitle, chapterUrl)
+
+
+        logger.debug("Story: <%s>", self.story)
+
+        return
+
+    def getChapterText(self, url):
+        """
+        Clean up a mcstories chapter page.
```

Comment:
Remove or correct references to mcstories please.

Reply:
OOps. :)